tests/queries/0_stateless/03040_dynamic_type_alters_1.reference

```diff
@@ -30,16 +30,16 @@ alter modify column 1
 0	0	\N	\N	\N	\N	\N
 1	1	\N	\N	\N	\N	\N
 2	2	\N	\N	\N	\N	\N
-3	3	3	3	\N	\N	\N
```

Comment:
Why has the order of the result columns changed? I don't see any changes related to that.

Reply:
The order didn't change, the actual output changed. It's now the same as with old analyzer.
The actual problem is how mutations of Dynamic column is handled in Memory engine. I will fix it in a separate PR. I didn't notice this probleb before because of the bug this PR fixes.

Reply:
https://github.com/ClickHouse/ClickHouse/pull/66066